lib/active_merchant/billing/gateways/sum_up.rb

```diff
@@ -0,0 +1,198 @@
+          MultiResponse.run do |r|
+            r.process { create_checkout(money, payment, options) }
+            r.process { complete_checkout(r.params['id'], payment, options) }
+          end
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Bearer )\w+), '\1[FILTERED]').
+          gsub(%r(("pay_to_email\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("number\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("cvv\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+      end
+
+      private
+
+      def create_checkout(money, payment, options)
+        post = {}
+
+        add_merchant_data(post, options)
+        add_invoice(post, money, options)
+        add_address(post, options)
+        add_customer_data(post, payment, options)
+
+        commit('checkouts', post)
+      end
+
+      def complete_checkout(checkout_id, payment, options = {})
+        post = {}
+
+        add_payment(post, payment, options)
+
+        commit('checkouts/' + checkout_id, post, :put)
+      end
+
+      def add_customer_data(post, payment, options)
+        post[:customer_id]      = options[:customer_id]
+        post[:personal_details] = {
+          email:      options[:email],
+          first_name: payment&.first_name,
+          last_name:  payment&.last_name,
+          tax_id:     options[:tax_id]
+        }
+      end
+
+      def add_merchant_data(post, options)
+        # Required field: pay_to_email
+        # Description: Email address of the merchant to whom the payment is made.
+        post[:pay_to_email] = @options[:pay_to_email]
+      end
+
+      def add_address(post, options)
+        post[:personal_details] ||= {}
+        post[:personal_details][:address] = {
+          city:        options[:billing_address][:city],
+          state:       options[:billing_address][:state],
+          country:     options[:billing_address][:country],
+          line_1:      options[:billing_address][:address1],
+          postal_code: options[:billing_address][:zip]
+        }
+      end
+
+      def add_invoice(post, money, options)
+        post[:checkout_reference] = options[:order_id]
+        post[:amount]             = amount(money)
+        post[:currency]           = options[:currency] || currency(money)
+        post[:description]        = options[:description]
+      end
+
+      def add_payment(post, payment, options)
+        post[:payment_type] = options[:payment_type] if PAYMENT_TYPES.include?(options[:payment_type])
+
+        credit_card = add_credit_card(payment)
+        post[:card] = {
+          name:         credit_card.name,
+          number:       credit_card.number,
+          expiry_month: format(credit_card.month, :two_digits),
+          expiry_year:  credit_card.year,
+          cvv:          credit_card.verification_value
+        }
+      end
+
+      def add_credit_card(payment)
```

Comment:
❓ Question:

What is the difference between the `add_credit_card` method a just calling `clone` on the payment object?

Reply:
It makes sense, I hadn't realized xD